src/main/java/com/dell/isg/smi/service/device/discovery/manager/threads/DeviceIdentificationThread.java

```diff
@@ -14,107 +17,121 @@
+			logger.trace(ipAddress + " Identified has Server Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.CHASSIS)) {
+			logger.trace(ipAddress + " Identified has Chassis Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.SWITCH)) {
+			logger.trace(ipAddress + " Identified has Switch Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.STORAGE)) {
+			logger.trace(ipAddress + " Identified has Storage Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.IOM)) {
+			logger.trace(ipAddress + " Identified has IOM Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.VM)) {
+			logger.trace(ipAddress + " Identified has VM Type: " + discoverDeviceInfo.getDeviceType());
+		} else {
+			discoverDeviceInfo.setDeviceType(DiscoveryDeviceTypeEnum.UNKNOWN.value());
+			discoverDeviceInfo.setStatus(DiscoveryDeviceStatus.UNKNOWN.name());
+		}
+	}
+
+	private boolean amI(String ipAddress, DiscoveryDeviceGroupEnum deviceGroupEnum) {
+		boolean isDeviceIdentified = false;
+		List<DiscoveryRule> discoveryRuleList = discoveryDeviceConfigProvider
+				.getAllDiscoveryRuleByGroup(deviceGroupEnum);
+		for (DiscoveryRule discoveryRule : discoveryRuleList) {
+			String protocol = discoveryRule.getProtocol();
+			String command = discoveryRule.getCommand();
+			List<DeviceType> enabledDeviceTypeList = (List<DeviceType>) CollectionUtils
+					.select(discoveryRule.getDeviceType(), predicateDiscoveryEnabled());
+			if (!CollectionUtils.isEmpty(enabledDeviceTypeList)) {
+				isDeviceIdentified = identify(command, DiscoveryDeviceProtocolEnum.fromValue(protocol),
+						enabledDeviceTypeList);
+			}
+
+		}
+		return isDeviceIdentified;
+	}
+
+	private Predicate<DeviceType> predicateDiscoveryEnabled() {
+		return new Predicate<DeviceType>() {
+			@Override
+			public boolean evaluate(DeviceType deviceType) {
+				if (deviceType == null) {
+					return false;
+				}
+				return deviceType.getEnabled() == Boolean.TRUE;
+			}
+		};
+	}
+
+	private boolean identify(String command, DiscoveryDeviceProtocolEnum protocol,
+			List<DeviceType> enabledDeviceTypeList) {
+		boolean isDeviceIdentified = false;
+		switch (protocol) {
+		case HTTP:
+		case HTTPS:
+			DiscoverDeviceTypeUtil.processHttps(discoverDeviceInfo, command, enabledDeviceTypeList);
+			break;
+		case TCP:
+			DiscoverDeviceTypeUtil.processTcp(discoverDeviceInfo, command, enabledDeviceTypeList);
+			break;
+		case SSH:
+			DiscoverDeviceTypeUtil.processSsh(discoverDeviceInfo, command, enabledDeviceTypeList);
+			break;
+		case TFTP:
+			DiscoverDeviceTypeUtil.processTftp(discoverDeviceInfo, command, enabledDeviceTypeList);
+			break;
+		}
+		if (!StringUtils.equals(discoverDeviceInfo.getDeviceType(), DiscoveryDeviceTypeEnum.UNKNOWN.name())) {
```

Comment:
discoverDeviceInfo.getDeviceType() is populated with enum values so better just use 

return 
      discoverDeviceInfo.getDeviceType() != DiscoveryDeviceTypeEnum.UNKNOWN.name()